### PR TITLE
feat: my account my profile page

### PR DESCRIPTION
**Pull request by darocha**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

### CODE REVIEW

File: packages/composables/src/useUser/params.ts [hunk 1]
```diff
@@ -2,39 +2,70 @@ import { UseUserFactoryParams, Context } from '@vue-storefront/core';
...
 const params: UseUserFactoryParams<User, UpdateParams, RegisterParams> = {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
+  provide() {
+    return {
+      cart: useCart()
+    };
+  },
+
   load: async (context: Context): Promise<User> => {
```
Comment (scernisevs) [inline packages/composables/src/useUser/params.ts:27]: hi @darocha ,

I was wondering if we could please brake down the props file for each action to be separate and add unit tests for each action as well? :)

**Review by scernisevs — COMMENTED** [review]
hi @darocha ,

Thank you very much for the PR! it looks great :) 

I've left a couple of comments and was wondering if we could please aim to keep the PR at around 500 lines (this one is 1000).  it might be little bit easier to review it if its broken down

File: packages/composables/src/useUser/params.ts [hunk 2]
```diff
@@ -2,39 +2,70 @@ import { UseUserFactoryParams, Context } from '@vue-storefront/core';
...
-  logIn,
+  logIn: async (context: Context, userParams: UseUserLoginParams): Promise<User> => {
+    if (userParams.username && userParams.password) {
+      const loginResponse: UseUserLoginResponse = await logIn(context, {username: userParams.username, password: userParams.password});
+      const { id } = loginResponse;
+      const { data } = await context.$bigcommerce.api.getCustomer({id});
+      return data[0];
+    }
+
+    return null;
```
Comment (zoltantakacs-born) [inline packages/composables/src/useUser/params.ts:62]: the logIn function what is implemented already (`packages/composables/src/useUser/actions/login.ts`) is good, the only modification needed there to get the customer data and return that in that function (and modified the type of the return value)

**Review by zoltantakacs-born — CHANGES_REQUESTED** [review]

File: packages/composables/src/useUser/actions/register.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { Context, UseUserRegisterParams } from '../../types';
+import { UseUserRegisterParams } from '../../types';
 import type { User } from '@vue-storefront/bigcommerce-api';
-
+import { Context } from '@vue-storefront/core';
```
[the remark this answers is not in the document]
Comment (darocha) [reply packages/composables/src/useUser/actions/register.ts]: No problem, I changed the code to use the custom context.

File: packages/composables/src/useUser/params.ts (hunk 1 above)
Comment (darocha) [reply packages/composables/src/useUser/params.ts:26]: Sure, I moved them to a separated file and added tests as well.

File: packages/composables/src/useUser/params.ts (hunk 2 above)
Comment (darocha) [reply packages/composables/src/useUser/params.ts]: Hi Zoltan. it is already implemented. The login function now return a user object.

**Review by WojtekTheWebDev — CHANGES_REQUESTED** [review]
Hi @darocha good job! There are few pages added to theme, probably from the `_theme` directory. can you add them to `theme/pages` in separate PR as they are in `_theme` to see what's changed here?

File: packages/theme/composables/useUserData/index.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+function getEmailAddress(user: User): string {
+  return user?.email;
+}
+
+export const userHelpers = {
+  getFirstName,
+  getLastName,
+  getFullName,
+  getEmailAddress
+};
```
Comment (zoltantakacs-born) [inline packages/theme/composables/useUserData/index.ts:24]: it is not bad, but not really a composable this way 😄

File: packages/theme/middleware/is-authenticated.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { COOKIE_KEY_CUSTOMER_DATA } from '@vue-storefront/bigcommerce-api/';
+
+export default async context => {
+  const customerData = context.$cookies.get(COOKIE_KEY_CUSTOMER_DATA);
+  if (!customerData) {
+    context.app.router.push('/');
+    context.redirect('/');
```
Comment (zoltantakacs-born) [inline packages/theme/middleware/is-authenticated.js:7]: is both needed? why?

**Review by zoltantakacs-born — CHANGES_REQUESTED** [review]

File: packages/theme/composables/useUserData/index.ts (hunk 4 above)
Comment (darocha) [reply packages/theme/composables/useUserData/index.ts:24]: you are correct. I modified to follow the same pattern as the other helpers.. Thanks :)

File: packages/theme/middleware/is-authenticated.js (hunk 5 above)
Comment (darocha) [reply packages/theme/middleware/is-authenticated.js:7]: No, there should only be one. I removed context.app.router.push('/')

File: packages/api-client/__tests__/api/customers/get.spec.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import { GetCustomersParameters, GetCustomersResponse } from '../../../src';
+
+describe('[bigcommerce-api-client] getCustomer', () => {
+  it('gets customer by id', async () => {
+    const getMock = jest.fn();
+    contextMock.client = {
+      get: getMock
+    };
+
+    contextMock.client.get.mockImplementationOnce(() => Promise.resolve());
```
[the remark this answers is not in the document]
Comment (darocha) [reply packages/api-client/__tests__/api/customers/get.spec.ts:12]: Good catch Zoltan. I removed the lines above.

**Pull request merged**